src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestResolversIdentitiesDataStore.java

```diff
@@ -0,0 +1,129 @@
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestResolversIdentitiesDataStore>
+    {
+
+        @NotNull
+        @Override
+        public TypeToken<? extends StandardRequestResolversIdentitiesDataStore> getFactoryOutputType()
+        {
+            return TypeToken.of(StandardRequestResolversIdentitiesDataStore.class);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends FactoryVoidInput> getFactoryInputType()
+        {
+            return TypeConstants.FACTORYVOIDINPUT;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestResolversIdentitiesDataStore getNewInstance(
+          @NotNull final IFactoryController factoryController, @NotNull final FactoryVoidInput factoryVoidInput, @NotNull final Object... context) throws IllegalArgumentException
+        {
+            return new StandardRequestResolversIdentitiesDataStore();
+        }
+
+        @NotNull
+        @Override
+        public NBTTagCompound serialize(
+          @NotNull final IFactoryController controller, @NotNull final StandardRequestResolversIdentitiesDataStore standardRequestIdentitiesDataStore)
+        {
+            final NBTTagCompound systemCompound = new NBTTagCompound();
+
+            systemCompound.setTag(TAG_TOKEN, controller.serialize(standardRequestIdentitiesDataStore.getId()));
+            systemCompound.setTag(TAG_LIST, standardRequestIdentitiesDataStore.getIdentities().keySet().stream().map(token -> {
+                final NBTTagCompound mapCompound = new NBTTagCompound();
+
+                mapCompound.setTag(TAG_TOKEN, controller.serialize(token));
+                mapCompound.setTag(TAG_RESOLVER, controller.serialize(standardRequestIdentitiesDataStore.getIdentities().get(token)));
+
+                return mapCompound;
+            }).collect(NBTUtils.toNBTTagList()));
+
+            return systemCompound;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestResolversIdentitiesDataStore deserialize(@NotNull final IFactoryController controller, @NotNull final NBTTagCompound nbt)
+        {
+            final IToken<?> token = controller.deserialize(nbt.getCompoundTag(TAG_TOKEN));
+            final NBTTagList list = nbt.getTagList(TAG_LIST, Constants.NBT.TAG_COMPOUND);
+
+            final Map<IToken<?>, IRequestResolver<?>> map = NBTUtils.streamCompound(list).map(nbtTagCompound -> {
+                final IToken<?> id = controller.deserialize(nbtTagCompound.getCompoundTag(TAG_TOKEN));
+                final IRequestResolver<?> resolver = controller.deserialize(nbtTagCompound.getCompoundTag(TAG_RESOLVER));
+
+                return new Tuple<IToken<?>, IRequestResolver<?>>(id, resolver);
+            }).collect(Collectors.toMap((Tuple<IToken<?>, IRequestResolver<?>> t) -> t.getFirst(), (Tuple<IToken<?>, IRequestResolver<?>> t) -> t.getSecond()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)